<?xml version="1.0"?>
<launch>
    <!-- true if running the nodes in the gazebo simulator environment. false if running on the real platform-->
    
    <!-- Stable version: Use time horizon of 2 second, 10 control nodes and control @ 50 Hz (control_freq_factor=5) -->
    <arg name="t_horizon" default="2"/>
    <arg name="n_nodes" default="40"/>
    <arg name="traj_resample_vel" default="true"/>

    <group ns="gp_mpc">
            <!-- Node that runs the MPC -->
            <node pkg="ros_gp_mpc" type="gp_ad_mpc_node.py" name="mpc_wrapper" output="screen">
                <rosparam file="$(find ros_gp_mpc)/config/simulation_run.yaml"/>
                <param name="t_horizon" value="$(arg t_horizon)"/>
                <param name="n_nodes" value="$(arg n_nodes)"/>                
                <param name="traj_resample_vel" value="$(arg traj_resample_vel)"/>    
            </node>
          
    </group>
</launch>